contracts/src/GatewayDiamond.sol

```diff
@@ -13,14 +13,15 @@ import {LibGateway} from "./lib/LibGateway.sol";
 import {SubnetID} from "./structs/Subnet.sol";
 import {LibStaking} from "./lib/LibStaking.sol";
 import {BATCH_PERIOD, MAX_MSGS_PER_BATCH} from "./structs/CrossNet.sol";
+import {Ownable} from "./Ownable.sol";
 
 error FunctionNotFound(bytes4 _functionSelector);
 
 bool constant FEATURE_MULTILEVEL_CROSSMSG = false;
 bool constant FEATURE_GENERAL_PUPRPOSE_CROSSMSG = true;
 uint8 constant FEATURE_SUBNET_DEPTH = 2;
 
-contract GatewayDiamond {
+contract GatewayDiamond is Ownable {
```

Comment:
Don't we need to add the Ownable methods to the diamond function selectors somehow? Otherwise they might not be externally invokable?

Reply:
fyi here's how the diamond Upgrade determines if the user making the request is the "contract owner" https://github.com/consensus-shipyard/ipc/blob/b718ddb2dc8f8d849f61cb9992ac7b565e03082d/contracts/src/diamond/DiamondCutFacet.sol#L19 It is using the method LibDiamond.enforceIsContractOwner(); 

the upgrade facet is helpful but it only adds in two methods - one that allows for explicitly changing the ownership (seems helpful and worth adding!) and the other returns the owner. I don't think we want to add is Ownable to the contract here because Owning happens in LibDiamond